src/os/bluestore/BlueStore.cc

```diff
@@ -3757,7 +3757,11 @@ int BlueStore::_open_alloc()
 {
   assert(alloc == NULL);
   assert(bdev->get_size());
-  alloc = Allocator::create(cct, cct->_conf->bluestore_allocator,
+ 
+  string type;
+  type = cct->_conf->bluestore_allocator;
+  assert(type == "stupid" || type == "bitmap");
+  alloc = Allocator::create(cct, type,
```

Comment:
could just check the return value of `Allocator::create()`, no need to check `type` beforehand.